### PR TITLE
Add keyboard shortcut to save automation/scene/script

### PR DESCRIPTION
**Pull request by gilsonmandalogo**
## Proposed change

Add CMD+S or CTRL+S keyboard shortcut to save automation/scene/script.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

No configuration needed.

## Additional information

- This PR is related to issue: #6397

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by iantrich — APPROVED** [review]

**Comment by iantrich** [comment]
🎉🎉🎉🎉

**Comment by bramkragten** [comment]
Can we make this a class mixing, that will call a function of the class when ctrl+s is called?

File: src/panels/config/automation/ha-automation-editor.ts [hunk 1]
```diff
@@ -80,6 +80,16 @@ export class HaAutomationEditor extends LitElement {
 
   @internalProperty() private _entityId?: string;
 
+  public connectedCallback() {
+    super.connectedCallback();
+    window.addEventListener("keydown", this._keydownEvent);
```
Comment (bramkragten) [inline src/panels/config/automation/ha-automation-editor.ts:85]: Can we add the listener to this element instead of window?

**Comment by gilsonmandalogo** [comment]
@bramkragten Thanks for your suggestions. This is my first time with LitElement, but I'll look at documentation to do this.

File: src/panels/config/automation/ha-automation-editor.ts (hunk 1 above)
Comment (gilsonmandalogo) [reply src/panels/config/automation/ha-automation-editor.ts]: Done.

File: src/mixins/keyboard-shortcut-mixin.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import { Constructor } from "../types";
+
+export const KeyboardShortcutMixin = <T extends Constructor<LitElement>>(
+  superClass: T
+) =>
+  class extends superClass {
+    private _keydownEvent = (event: KeyboardEvent) => {
+      if ((event.ctrlKey || event.metaKey) && event.key === "s") {
+        event.preventDefault();
+        this.saveCallback();
```
Comment (bramkragten) [inline src/mixins/keyboard-shortcut-mixin.ts:11]: I think the name should be a bit clearer, maybe
```suggestion
        this.handleKeyboardSave();
```

**Review by bramkragten — APPROVED** [review]
:tada: great!

File: src/mixins/keyboard-shortcut-mixin.ts (hunk 2 above)
Comment (gilsonmandalogo) [reply src/mixins/keyboard-shortcut-mixin.ts]: Done! Good suggestion.

**Comment by donkawechico** [comment]
First of all, this is awesome.

Second of all, and this is an admittedly unlikely scenario, but now that Quick Bar (with its two hotkeys) is a thing, we might want to think of ways to avoid collisions.

Some kind of shortcut registry, or at least unifying them all under a common mixin where it's a dev can see what's already taken.

**Pull request merged**